index.js

```diff
@@ -93,121 +114,153 @@ rl.on('line', function(line) {
-            request('POST', '/login', {'Content-Type': 'application/json',
-                                       'Content-Length': postData.length}, postData).then(result => {
-                console.log('Webtoken from login:', result.token);
-                var token = result.token;
-        
-                // Create group (admin user will be autmatically added to group)
-                var groupData = {
-                    name: groupName
-                };
-                console.log('Creating group ' + groupData.name + '.......');
-                jsonData = JSON.stringify(groupData);
-                request('POST', '/groups', {'Content-Type': 'application/json',
-                                    'Content-Length': jsonData.length,
-                                    'Authorization': 'Bearer ' + token}, jsonData).then(result => {
-                    var groupId = result.id;
-                    console.log('group created, id:', groupId);
-        
-                    function handleCreateUser(user) {
-                        // Create users
-                        var postData = JSON.stringify(user);
-                        console.log('Creating user ' + user.forename + '......');
-                        return request('POST', '/users', {'Content-Type': 'application/json',
-                                                          'Content-Length': postData.length}, postData);
-                    }
-                    console.log('Creating users......');
-                    var createUserPromises = users.map(handleCreateUser);
-                    var userIdMap = new Map();
-                    Promise.all(createUserPromises).then(results => {
-                        results.forEach(result => {
-                            userIdMap.set(result.username, result.id);
-                            console.log('User created with username:', result.username, 'id:', result.id);
-                        });
+            console.log('Creating admin user', admin.forename, '......');
+            createUser(admin, masterToken).then(result => {
+                console.log('Admin user created with username:', result.username, 'id:', result.id);
 
-                        function handleAddUserToGroup(user) {
-                            console.log('Adding user (' + user.forename + ') to group ' + groupData.name + '......');
-                            return request('PUT', '/groups/' + groupId + '/members/' + userIdMap.get(user.phoneNumber),
-                                {'Authorization': 'Bearer ' + token});
+                console.log('Logging in admin user......');
+                login(admin).then(result => {
+                    console.log('token for admin user:', result.token);
+                    token = result.token;
+
+                    console.log('Creating group ' + groupName, 'with token:', masterToken, '.......');
+                    createGroup(groupName, token).then(result => {
```

Comment:
not here though, because we need the admin token, so we need to stay in the promise callback where we have access to the admin token :)